pysmt/oracles.py

```diff
@@ -241,6 +241,12 @@ def walk_toreal(self, formula, args, **kwargs):
         return theory_out
         rtype = formula.symbol_name()
 
+    def walk_realtoint(self, formula, args, **kwargs):
+        #pylint: disable=unused-argument
+        """Extends the Theory with NIRA."""
+        theory_out = args[0].set_nira() # This makes a copy of args[0]
```

Comment:
This seems problematic. The output should extend the input with nonlinearity as well as integer arithmetic.